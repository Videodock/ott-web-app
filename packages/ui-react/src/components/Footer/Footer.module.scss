@use '@jwp/ott-ui-react/src/styles/variables';

.footer {
  padding: 20px 40px;
  line-height: variables.$base-line-height;
  letter-spacing: 0.15px;
  text-align: center;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);

  a,
  a:visited,
  a:active,
  a:hover {
    color: variables.$white;
    text-decoration: none;
  }
}

.list {
  margin: 0;
  padding: 0;
  list-style: none;

  li {
    display: inline-block;
    padding: 0 4px;

    &:not(:last-child)::after {
      content: '|';
      padding-left: 2px;
    }
  }
}

.testFixMargin {
  margin-bottom: 50px;
}
