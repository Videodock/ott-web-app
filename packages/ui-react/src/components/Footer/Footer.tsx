import React from 'react';
import classNames from 'classnames';
import { IS_DEVELOPMENT_BUILD } from '@jwp/ott-common/src/utils/common';

import MarkdownComponent from '../MarkdownComponent/MarkdownComponent';

import styles from './Footer.module.scss';

type Props = {
  text: string;
};

const Footer: React.FC<Props> = ({ text }) => {
  const chunks = text.split('|').map((chuck) => chuck.trim());
  const footerContent = chunks.map((value, index) => <MarkdownComponent key={index} markdownString={value} inline tag={chunks.length > 1 ? 'li' : 'div'} />);
  const wrapper = chunks.length > 1 ? <ul className={styles.list}>{footerContent}</ul> : footerContent;

  return (
    // The extra style below is just to fix the footer on mobile when the dev selector is shown
    <footer className={classNames(styles.footer, { [styles.testFixMargin]: IS_DEVELOPMENT_BUILD })}>{wrapper}</footer>
  );
};

export default Footer;
