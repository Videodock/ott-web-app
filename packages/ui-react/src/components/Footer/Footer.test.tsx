import React from 'react';
import { render } from '@testing-library/react';

import Footer from './Footer';

describe('<Footer>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Footer text="Simple" />);

    expect(container).toMatchSnapshot();
  });
  test('renders and matches snapshot without links', () => {
    const { container } = render(<Footer text="Text one | Text two" />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot with two links', () => {
    const { container } = render(<Footer text="Two links | [jwplayer.com](https://www.jwplayer.com/) | [jwplayer.com](https://www.jwplayer.com/)" />);

    expect(container).toMatchSnapshot();
  });
});
