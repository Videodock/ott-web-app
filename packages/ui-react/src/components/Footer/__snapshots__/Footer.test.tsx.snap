// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer> > renders and matches snapshot 1`] = `
<div>
  <footer
    class="_footer_33bf66 _testFixMargin_33bf66"
  >
    <div
      class="_markdown_e219e2"
    >
      Simple
    </div>
  </footer>
</div>
`;

exports[`<Footer> > renders and matches snapshot with two links 1`] = `
<div>
  <footer
    class="_footer_33bf66 _testFixMargin_33bf66"
  >
    <ul
      class="_list_33bf66"
    >
      <li
        class="_markdown_e219e2"
      >
        Two links
      </li>
      <li
        class="_markdown_e219e2"
      >
        <a
          href="https://www.jwplayer.com/"
          rel="noopener"
          target="_blank"
        >
          jwplayer.com
        </a>
      </li>
      <li
        class="_markdown_e219e2"
      >
        <a
          href="https://www.jwplayer.com/"
          rel="noopener"
          target="_blank"
        >
          jwplayer.com
        </a>
      </li>
    </ul>
  </footer>
</div>
`;

exports[`<Footer> > renders and matches snapshot without links 1`] = `
<div>
  <footer
    class="_footer_33bf66 _testFixMargin_33bf66"
  >
    <ul
      class="_list_33bf66"
    >
      <li
        class="_markdown_e219e2"
      >
        Text one
      </li>
      <li
        class="_markdown_e219e2"
      >
        Text two
      </li>
    </ul>
  </footer>
</div>
`;
